sdk/search/azure-search-documents/CHANGELOG.md

```diff
@@ -11,6 +11,7 @@
 
 - Renamed `SearchClient.speller` to `SearchClient.query_speller`.
 - Removed keyword arguments from `SearchClient`: `answers` and `captions`.
+- `SentimentSkill`, `EntityRecognitionSkill`: added client-side validation to prevent sending unsupported parameters.
```

Comment:
Let's double check with @johanste because in our guidelines, we have

```py
DO NOT validate service parameters. This includes null checks, empty strings, and other common validating conditions. Let the service validate any request parameters.
```

https://azure.github.io/azure-sdk/general_implementation.html

Does this conflict against our guideline?

Reply:
I agree @johanste should weigh in. The discussion we had with Anna was that we should validate because the service does not. Her issue was with validating in the constructor, believing we should instead validate in a way that works with dict input as well as strongly-typed models (which is what this does).

Reply:
Client side validation is acceptable when it prevents silent failures. Which seem to be the case here. If the service had done proper validation (and given a reasonable error message), client side validation would not have been needed.